src/Entity/Int32EntityId.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Wikibase\DataModel\Entity;
+
+use InvalidArgumentException;
+
+/**
+ * Interface for EntityIds that can be converted to a unique positive 32 bit integer in the range
+ * [1..2147483647], and back from the entity type and the number. Entity types that do not meet this
+ * criteria should not implement this interface.
+ *
+ * Entity types are not required and not guaranteed to implement this interface. Use the full string
+ * serialization whenever you can and avoid using numeric IDs.
+ *
+ * @since 6.1
+ *
+ * @license GPL-2.0+
+ * @author Thiemo Mättig
+ */
+interface Int32EntityId {
+
+	/**
+	 * @since 6.1
+	 *
+	 * @param string $idSerialization
+	 *
+	 * @throws InvalidArgumentException When the provided serialization does not allow getNumericId
+	 *  to return a unique positive 32 bit integer.
+	 */
+	public function __construct( $idSerialization );
+
+	/**
+	 * @since 6.1
+	 *
+	 * @return int Guaranteed to be a unique positive 32 bit integer.
```

Comment:
In order to use the return value from this method, I need to know that it's => 0, and <= 2147483647. Only then can I use it with our current DB schema with both MySQL and PostGres. That guarantee is the reason we have this interface. So that guarantee should be mentioned here explicitly.